src/ir/module.h

```diff
@@ -48,6 +50,7 @@ class Operation {
   const Block* parent() const { return parent_; }
   const NamedValueMap& inputs() const { return inputs_; }
   const NamedAttributeMap& attributes() const { return attributes_; }
+  const Module* module() const { return module_.get(); }
```

Comment:
I'm a little worried about null safety here, as it's pretty subtle that the implementation module not being present is just wrapped in a `std::unique_ptr`. I'm not sure I see solutions that are extremely better though.

Reply:
I agree with the overall sentiment around null safety. However, I don't think things are that much better if we return `const &`. The main advantage of returning `const &` is that we indicate to the caller that the value cannot be `null`. 

Other than I am not sure how better is it in terms of memory safety. The underlying object should still exist until the lifetime of the `const &`. It is true that there is automatic extension of the lifetime of the underlying object, but it is quite limited: https://abseil.io/tips/107

In this specific case, the API suggests that the `module()` could be `nullptr` and the callers have to take care of that situation appropriately. So, it seems OK to return `const Module*`?